File: merge_in_between_linked_lists.go
func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
    ptr1 := list1
    for i := 0; i < a-1; i++ {
        ptr1 = ptr1.Next
    }

    ptr2 := ptr1
    for i := 0; i < b-a+1; i++ {
        ptr2 = ptr2.Next
    }

    ptr1.Next = list2
    for list2.Next != nil {
        list2 = list2.Next
    }
    list2.Next = ptr2.Next

    return list1
}

// Time Complexity : O(n)
// Space Complexity : O(1)

File: merge_in_between_linked_lists.cpp
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode* ptr1 = list1;
        while(--a) {
            ptr1 = ptr1->next;
            b--;
        }

        ListNode* ptr2 = ptr1;
        while(b > -1) {
            ptr2 = ptr2->next;
            b--;
        }
        
        ptr1->next = list2;
        while(list2->next) {
            list2 = list2->next;
        }
        list2->next = ptr2;
        return list1;
    }
};

// Time Complexity : O(n)
// Space Complexity : O(1)